ecosystem/sep-0023.md

```diff
@@ -77,14 +79,20 @@ The following steps transform a binary key into a strkey:
 3.  If we are encoding a multiplexed address, append an 8-byte memo ID
     in network byte order (most significant byte first).
 
-4.  Compute a 16-bit CRC16 checksum of the combined version byte,
-    optional memo ID, and binary key (using polynomial
-    x<sup>16</sup> + x<sup>12</sup> + x<sup>5</sup> + 1). Append the
-    two-byte checksum to the result of the previous step (e.g.,
-    producing a 35-byte quantity for a non-multiplexed ED25519 public
-    key, or 43 byte quantity for a multiplexed one).
+4.  If we are encoding a signed payload, append a 4-byte length in
```

Comment:
Payloads are 64 bytes, so can we just use a single byte for length here?

Reply:
Today payloads are 64 bytes, but in the future they might be increased. We could reduce the number of bytes here as an optimization I guess to at least down to 2 bytes. I think originally @stanford-scs was trying to keep strkeys as generally the encoding / decoding to valid XDR, typically the full XDR for that type, but we haven't followed that strictly in the past. XDR requires a length to be 4 bytes which is where the 4 bytes comes from.

Reply:
I guess if we really wanted to optimize while preserving future possible used we could encode the length into 7-bit chunks before encoding into base32. It would make the minimum length of the length 1 byte. If we ever supported 32-bit lengths for the payload it the length of the length would be 5 bytes. This might be a pain to code in most SDKs, but most SDKs could probably punt on actually implementing it on the assumption payloads are only length 64 max today.

Reply:
I'm inclined to say nothing is worth optimizing here and leaving it as it is. Wdyt @stanford-scs @ire-and-curses?

Reply:
> I think originally @stanford-scs was trying to keep strkeys as generally the encoding / decoding to valid XDR, typically the full XDR for that type,

If this was an initial design goal then I'm not against keeping it that way (though muxed addresses already deviate from this, with the ID coming second in the StrKey but first in the XDR). I just wanted to bring up that we might be able to shave 2-3 bytes off per strkey here. 

Over-optimizing for base32 seems excessive, and dropping to 1 byte does limit future extensions (though having signed >256 byte payloads seems fictitious: you should hash your payload and sign _that_ instead), but 2 bytes would be a reasonable middle ground. However, as you noted, it would deviate from the underlying XDR, so if that's an important element to preserve here, this micro-optimization discussion probably isn't worth it.

Reply:
This is what the saving looks like:
```
4-byte len: PA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAQACAQDAQCQMBYIBEFAWDANBYHRAEISCMKBKFQXDAMRUGY4DUPB6IBZGM
3-byte len: PA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAEAAQEAYEAUDAOCAJBIFQYDIOB4IBCEQTCQKRMFYYDENBWHA5DYPSAOJT
2-byte len: PA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUABAAEBAGBAFAYDQQCIKBMGA2DQPCAIREEYUCULBOGAZDINRYHI6D4QDSMY
1-byte len: PA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUIABAIBQIBIGA4EASCQLBQGQ4DYQCEJBGFAVCYLRQGI2DMOB2HQ7EA4TG
```
Ref: https://go.dev/play/p/sLUfHJImdR4

A 2-byte length would reduce the total strkey by 3 characters.

Reply:
I'll be honest, the only reason why I even noticed this is because I saw the `AAAA` characters in the middle of the strkey and that seemed odd to me. So reducing the length has an aesthetic benefit as well :laughing: but again I don't have a strong opinion here, especially if we want adherence to XDR, as it'd make SDK code simpler.

Reply:
Unless there's a scaling implication, such as a significant cost to store the strkey in databases, I don't think we should over optimize it. The XDR type could allow a larger value in the future.

Reply:
Eiiiii

Reply:
Good discussion, but yeah I also don't think this is worth optimising for.